mycore-iview2/src/main/java/org/mycore/iview2/frontend/MCRPDFTools.java

```diff
@@ -30,33 +30,63 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.PDPage;
+import org.apache.pdfbox.pdmodel.common.PDDestinationOrAction;
+import org.apache.pdfbox.pdmodel.interactive.action.PDActionGoTo;
+import org.apache.pdfbox.pdmodel.interactive.documentnavigation.destination.PDDestination;
+import org.apache.pdfbox.pdmodel.interactive.documentnavigation.destination.PDPageDestination;
 import org.apache.pdfbox.rendering.PDFRenderer;
 import org.mycore.common.content.MCRContent;
 import org.mycore.tools.MCRPNGTools;
 
 /**
  * @author Thomas Scheffler (yagee)
  */
-class MCRPDFTools implements AutoCloseable {
+public class MCRPDFTools implements AutoCloseable {
 
-    static final int PDF_DEFAULT_DPI = 72; //from private org.apache.pdfbox.pdmodel.PDPage.DEFAULT_USER_SPACE_UNIT_DPI
+    static final int PDF_DEFAULT_DPI = 72; // from private org.apache.pdfbox.pdmodel.PDPage.DEFAULT_USER_SPACE_UNIT_DPI
 
-    private static Logger LOGGER = LogManager.getLogger(MCRPDFTools.class);
+    private static final Logger LOGGER = LogManager.getLogger(MCRPDFTools.class);
 
-    private MCRPNGTools pngTools;
+    private final MCRPNGTools pngTools;
 
     MCRPDFTools() {
         this.pngTools = new MCRPNGTools();
     }
 
-    static BufferedImage getThumbnail(Path pdfFile, int thumbnailSize, boolean centered) throws IOException {
+    /**
+     * The old method did not take the thumbnail size into account, if centered =
+     * false;
+     *
+     * @see #getThumbnail(int, Path, boolean)
+     */
+    @Deprecated
+    public static BufferedImage getThumbnail(Path pdfFile, int thumbnailSize, boolean centered) throws IOException {
+        return getThumbnail(-1, pdfFile, false);
+    }
+
+    /**
+     * This method returns a Buffered Image as thumbnail if an initial page was set,
+     * it will be return - if not the first page
+     *
+     * @param thumbnailSize - the size: size = max(width, height) 
+     *                        a size &lt; 0 will return the original size and centered parameter will be ignored
+     * @param pdfFile       - the file from which the thumbnail will be taken
+     * @param centered      - if true, a square (thumbnail with same width and
+     *                      height) will be returned
+     * @return a BufferedImage as thumbnail
+     *
+     * @throws IOException
+     */
+    public static BufferedImage getThumbnail(int thumbnailSize, Path pdfFile, boolean centered) throws IOException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'getThumbnail(int, Path, boolean)' has an NPath complexity of 417](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5425159)